### PR TITLE
SALTO-5922 use provided type name for recurseInto types instead of generating default ducktype nested type name

### PR DESCRIPTION
**Pull request by netama**
When adding a field with `resource.recurseInto`, the user provides the name of the type they are creating, and various downstream places rely on it. this was ignored when generating the ducktype type name, causing non-standalone types to not be defined correctly (and e.g. miss field customizations and not support defining nested references)

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/adapter-components/test/fetch/fetch.test.ts [hunk 1]
```diff
@@ -89,6 +89,7 @@ describe('fetch', () => {
     it('should generate elements correctly', async () => {
       const customSaltoError: SaltoError = {
         message: 'error fetching default option',
+        detailedMessage: 'error fetching default option',
```
Comment (netama) [inline packages/adapter-components/test/fetch/fetch.test.ts:92]: unrelated

**Review by shir-reifenberg — APPROVED** [review]
LGTM 😄 
are there any real scenarios where we use `recurseInto` without `standalone`? if so we should verify this change doesn't break any references

**Comment by netama** [comment]
it can only add new references because currently these fields have type `list<unknown>` and are not traversed when searching for references (this is why I needed this fix now)
there does seem to be a broken rule in [confluence](https://github.com/salto-io/salto/blob/main/packages/confluence-adapter/src/definitions/references.ts#L40) that can now be fixed to use the correct part type and actually include the references, but will do that separately
